pkg/application/application.go

```diff
@@ -1,42 +1,124 @@
+// If application already exists, this errors out.
 // If no project is set, this errors out.
 func Create(name string) error {
-	if err := occlient.CreateNewProject(name); err != nil {
-		return errors.Wrapf(err, "unable to create application: %v", name)
+	// TODO: right now this assumes that there is a current project in openshift
+	// when we have project support in ocdev, this should call project.GetCurrent()
+	// TODO: use project abstaction
+	project, err := occlient.GetCurrentProjectName()
+	if err != nil {
+		return errors.Wrap(err, "unable to create new application")
 	}
+
+	cfg, err := config.New()
+	if err != nil {
+		return errors.Wrap(err, "unable to create new application")
+	}
+
+	err = cfg.SetActiveApplication(name, project)
+	if err != nil {
+		return errors.Wrap(err, "unable to create new application")
+	}
+
 	return nil
 }
 
-func List() (string, error) {
-	project, err := occlient.GetProjects()
+// List all application in current project
+func List() ([]ApplicationInfo, error) {
+	// TODO: use project abstaction
+	project, err := occlient.GetCurrentProjectName()
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list applications")
+	}
+
+	appNames, err := occlient.GetLabelValues(project, ApplicationLabel)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to list applications")
+	}
+
+	activeApplication, err := GetCurrent()
 	if err != nil {
-		return "", errors.Wrap(err, "unable to list applications")
+		return nil, errors.Wrap(err, "unable to list applications")
 	}
-	return project, nil
+
+	applications := []ApplicationInfo{}
+
+	for _, name := range appNames {
+		active := false
+		if activeApplication == name {
+			active = true
+		}
+		applications = append(applications, ApplicationInfo{
+			Name:    name,
+			Active:  active,
+			Project: project,
+		})
+	}
+
+	return applications, nil
 }
 
 // Delete deletes the given application
 func Delete(name string) error {
-	if err := occlient.DeleteProject(name); err != nil {
-		return errors.Wrapf(err, "unable to delete application: %v", name)
-	}
-	return nil
+	// if err := occlient.DeleteProject(name); err != nil {
+	// 	return errors.Wrapf(err, "unable to delete application: %v", name)
+	// }
+	// TODO: implement
+	return fmt.Errorf("NOT IMPLEMENTED")
 }
 
+// GetCurrent application if no application is active it returns defaultApplication name
 func GetCurrent() (string, error) {
-	app, err := occlient.GetCurrentProjectName()
+	// TODO: use project abstaction
+	project, err := occlient.GetCurrentProjectName()
+	if err != nil {
+		return "", errors.Wrap(err, "unable to get active application")
+	}
+
+	cfg, err := config.New()
 	if err != nil {
-		return "", errors.Wrap(err, "unable to get the active application")
+		return "", errors.Wrap(err, "unable to get active application")
+	}
+
+	app := cfg.GetActiveApplication(project)
+	// if no Application is active use default
+	if app == "" {
+		app = defaultApplication
```

Comment:
I don't think this should be the behavior. If no application is set, then we should return an empty string. We don't know if the default application was created or not, this might bring in inconsistency in the code because this will be called at a lot of places in the code.
This logic should be outside `GetCurrent()`

Reply:
But that will mean that you will have logic for default application in multiple places :-(

You will have to handle default application after every `GetCurrent` call anyway. Why not have it inside? I think that this makes much more sense.

Reply:
But then this is not the correct information :( The current application is _not_ `defaultApp`, it should be set as `defaultApp` outside this function.
It does not have to be set everywhere, instead like it's currently in master, it's set once in `cmd`, and then we pass the application name to all functions instead of calling `GetCurrent()` everywhere,

Reply:
I can't be as it is in the master.
Because following has to be possible.
```
# no application exists or is active
ocdev component create nodejs
```
(use case 19)

Before it was possible because we couldn't have no active application because  application was project.
But not this state is possible.
That means that everytime you call `GetCurrent` you have to also handle state where there is no application.
And this can't lead to error.

Reply:
okay